PREREQUISITE.md

```diff
@@ -183,76 +183,74 @@ The sample code below is an example of adding permission through the UI. Please
+                new ActivityResultCallback<Boolean>() {
+                    @Override
+                    public void onActivityResult(Boolean result) {
+                        if (result) {
+                            // PERMISSION GRANTED
+                        } else {
+                            // PERMISSION DENIED
+                        }
+                    }
+                });
+
         button = findViewById(R.id.button2);
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                int permission = ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO);
-
-                if (permission == PackageManager.PERMISSION_DENIED) {
-                    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                        requestPermissions(new String[]{
-                                Manifest.permission.RECORD_AUDIO}, 1000);
-                    }
+                if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    final String permission = Manifest.permission.RECORD_AUDIO;
+                    requestPermissionLauncher.launch(permission);
                 }
             }
         });
     }
-```
 
-```java
-   //after requestPermission
-    @Override
-    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
-        if(requestCode == 1000) {
-            boolean check_result = true;
-            for (int result : grantResults) {
-                if (result != PackageManager.PERMISSION_GRANTED) {
-                    check_result = false;
-                    break;
-                }
-            }
-            if(check_result == true) {
-
-            } else {
-                finish();
-            }
-        }
-    }
 }
 ```
 
 This is `MainActivity` for adding permission.
 
 ```java
-int permission = ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO);
-
-if (permission == PackageManager.PERMISSION_DENIED) {
-    if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-        requestPermissions(new String[]{
-                Manifest.permission.RECORD_AUDIO}, 1000);
-    }
+ActivityResultLauncher requestPermissionLauncher = registerForActivityResult(
+        new ActivityResultContracts.RequestPermission(),
+        new ActivityResultCallback<Boolean>() {
+            @Override
+            public void onActivityResult(Boolean result) {
+                if (result) {
+                    // PERMISSION GRANTED
+                } else {
+                    // PERMISSION DENIED
+                }
+            }
+        });
+if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+    final String permission = Manifest.permission.RECORD_AUDIO;
+    requestPermissionLauncher.launch(permission);
```

Comment:
I use ActivityResultLauncher instead of requestPermissions.
I'm not sure this is best way, so please tell me if there are other good ways.